core/vm/interpreter.go

```diff
@@ -132,7 +128,11 @@ func NewEVMInterpreter(evm *EVM, cfg Config) *EVMInterpreter {
 // considered a revert-and-consume-all-gas operation except for
 // ErrExecutionReverted which means revert-and-keep-gas-left.
 func (in *EVMInterpreter) Run(contract *Contract, input []byte, readOnly bool) (ret []byte, err error) {
-	if contract.Address() == BuiltinAddr {
+	// Deprecate special handling of [BuiltinAddr] as of SubnetEVM.
+	// In SubnetEVM, the contract deployed in the genesis is overridden by a deprecated precompiled
+	// contract which will return an error immediately if its ever called. Therefore, this function should
+	// never be called after SubnetEVM with [BuiltinAddr] as the contract address.
+	if !in.evm.chainRules.IsSubnetEVM && contract.Address() == BuiltinAddr {
```

Comment:
I'm not sure we actually need this check because we never deploy the deprecated precompiled contract